django/middleware/csrf.py

```diff
@@ -229,7 +231,7 @@ def _get_token(self, request):
                 request.META['CSRF_COOKIE_NEEDS_UPDATE'] = True
             return csrf_token
 
-    def _set_token(self, request, response):
+    def _set_csrf_cookie(self, request, response):
```

Comment:
Is this the right name? If I'm using `settings.CSRF_USE_SESSIONS` then no cookie gets set.

Reply:
I'm in two minds on this. On one hand, yes, @carltongibson is right. On the other hand, a lot of this code treats `CSRF_USE_SESSIONS` as an afterthought (which, historically, it is, but...). Maybe this way is more consistent with the rest, and changing the not-necessarily-correct references to cookies (e.g. `request.META['CSRF_COOKIE']`) into something more befitting the dual use should be the subject of a separate refactoring, if we want it.

Perhaps when we remove the distinction between the cookie/token and the secret, we should rename this to `_set_secret()`. Perhaps a good name now is `_set_masked_secret()`. Not sure.

Whatever choice is picked here, should also apply to `_add_new_csrf_cookie()` above.

Reply:
Thanks for starting to take a look, @carltongibson. The `csrf_cookie` in the name here is meant to be a reference to `request.META['CSRF_COOKIE']`, which is the value getting saved / stored (or the value being used to set). What about one of these instead? 

* `_save_csrf_cookie()`
* `_store_csrf_cookie()`

That might make clearer that `csrf_cookie` refers to _what_ is getting stored (the value) instead of _where_ it's getting stored. The main / only(?) downside I see with these alternatives IMO is that, at least when `settings.CSRF_USE_SESSIONS` is false, one doesn't normally think of setting a cookie as saving / storing something.

Also note that the call to this function comes immediately after the line checking whether `request.META['CSRF_COOKIE_NEEDS_UPDATE']` is true.

Some related, possibly helpful snippets from the current CSRF docs:

The ["How it works" section](https://docs.djangoproject.com/en/3.2/ref/csrf/#how-it-works) starts out with the following language, even though (just like your comment) it's only correct when `CSRF_USE_SESSIONS` is false (it refers to a "CSRF cookie" and uses the "cookie is set" terminology): 

> The CSRF protection is based on the following things:
>
> 1. A CSRF cookie that is based on a random secret value, which other sites will not have access to. This cookie is set by CsrfViewMiddleware. It is sent with every response ...

The [`CSRF_USE_SESSIONS` docs](https://docs.djangoproject.com/en/3.2/ref/settings/#std:setting-CSRF_USE_SESSIONS) use the "store" terminology I was suggesting as an alternative above. It uses it even in the case where `CSRF_USE_SESSIONS` is false:

> Whether to store the CSRF token in the user’s session instead of in a cookie. It requires the use of `django.contrib.sessions`. Storing the CSRF token in a cookie (Django’s default) is safe, but storing it in the session is common practice in other web frameworks ...

Reply:
> Perhaps a good name now is _set_masked_secret(). Not sure.

Thanks, @shaib. Yes, that would probably be a better name in the current state. One of my goals here, though, was to switch to a better name that would also have the property of being consistent with the upcoming code change. That way, the future diff will be smaller / easier for people to read because it will have less renamings, etc. happening at the same time.